src/geb/trans.lisp

```diff
@@ -189,3 +189,178 @@ should calculate
+with"
+  (let ((lft (dom (mcar obj)))
+        (rt  (dom (mcadr obj)))
+        (cpr (dom obj)))
+    (cond ((< (length (seqwidth lft))
+              (length (seqwidth rt)))
+           (seqn:branch-seq
+            (seqn:composition (to-seqn (mcar obj))
+                              (seqn:composition
+                               (seqn:remove-right (seqwidth lft))
+                               (seqn:parallel-seq (drop-width (butlast
+                                                               (cdr (seqwidth cpr))
+                                                               (- (length (seqwidth rt))
+                                                                  (length (seqwidth lft))))
+                                                              (seqwidth lft))
+                                                  (drop-nil (nthcdr (1+ (length (seqwidth lft)))
+                                                                    (seqwidth cpr))))))
+            ;; branch on the left counts the first entries of an occuring and drops the rest
+            ;; then injects it into the smaller bit sizes if necessary
+            ;; (max(a1, b1),,,.,(max (an, bn),..., max(0, bm))) -> (a1,...,an,0)
+            ;; -> (a1,....,an)
+            (composition (to-seqn (mcadr obj))
+                         (drop-width (cdr (seqwidth cpr)) (seqwidth rt)))))
+          ;; branch on the right is just inejcting
+          ;; (max(a1, b1),..., max(0, bn)) -> (b1, ...., bn) -> codom))
+          ((< (length (seqwidth rt))
+              (length (seqwidth lft)))
+           (branch-seq (composition (to-seqn (mcar obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth lft)))
+                       (composition (to-seqn (mcadr obj))
+                                    (composition (remove-right (seqwidth rt))
+                                                 (parallel-seq (drop-width (butlast
+                                                                            (cdr (seqwidth cpr))
+                                                                            (- (length (seqwidth lft))
+                                                                               (length (seqwidth rt))))
+                                                                           (seqwidth rt))
+                                                               (drop-nil (nthcdr (1+ (length (seqwidth rt)))
+                                                                                 (seqwidth cpr))))))))
+          ((= (length (seqwidth rt))
+              (length (seqwidth lft)))
+           (branch-seq (composition (to-seqn (mcar obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth lft)))
+                       (composition (to-seqn (mcadr obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth rt))))))))
```

Comment:
Abstract this, I can't read this just by looking. I'm sure you can find similarities and remove it. Too much code.